src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -2056,9 +2056,18 @@ QSet<QVariant> QgsPostgresProvider::uniqueValues( int index, int limit ) const
   {
     // get the field name
     QgsField fld = field( index );
-    QString sql = QStringLiteral( "SELECT DISTINCT %1 FROM %2" )
-                  .arg( quotedIdentifier( fld.name() ),
-                        mQuery );
+    // Postgres does not provide an "index skip scan" feature.
+    // This can make SELECT DISTINCT queries really slow.
+    // This query emulates the "index skip scan" behavior to quickly retrieve unique values
+    QString sql = QStringLiteral(
```

Comment:
```suggestion
    const QString sql = QStringLiteral(
```

Reply:
This won't work because the `sql` is modified later.